### ISSUE TITLE
Don't send emails in the request thread

### ISSUE THREAD

**Issue by dchiquito** [dandi/dandi-api#597]
As it currently stands, we generally send emails directly from view methods. If something were to go wrong with the email sending, the API request would fail, even though the operation completed successfully. We should either delegate the sending of the email to a task or wrap the sending of the email in a try/catch to ensure it never disrupts the request.

**Comment by waxlamp** [comment]
Delegating to a celery task feels too heavy. Wrapping in try/catch and sending errors to Sentry (but letting the request succeed) would be a good approach.

**Comment by danlamanna** [comment]
Yeah I think we should move email sending to celery tasks, but it's low priority.
